codes/classical/properties/random.yml

```diff
@@ -6,15 +6,43 @@
 code_id: random
 
 name: 'Random code'
-#introduced: ''
+introduced: '\cite{doi: 10.1109/TIT.2002.800480}'
 
-description: 'Code whose construction is non-deterministic in some way, i.e., codes that utilize an elements of randomness somewhere in their construction. Members of this class range from fully non-deterministic codes, to codes whose multi-step construction is deterministic with the exception of a single step.'
+description: 'Code whose construction is non-deterministic in some way, i.e., codes that utilize an elements of randomness somewhere in their construction. Members of this class range from fully non-deterministic codes, to codes whose multi-step construction is deterministic with the exception of a single step.
+              Typically, random codes are selected with uniform distribution from a ensemble of codes. Examples are random binary codes and random binary linear codes.
+              A random binary code is a set of 2^{K}=2^{NR} codewords with length N. Each bit in the codeword is randomly chosen between 0 and 1 with equal probability.
+              A random binary linear code is generated from a random chosen K generators of length N where each bit of the generators is randomly chosen between 0 and 1 with equal probability.
+              Equivalently, A random binary linear code is defined by a randomly generated K by N generator matrix where each entry is randomly chosen between 0 and 1 with equal probability.'
 
 #notes:
 #  - ''
 
+protection: 'The code can protect against random errors, e.g. from Binary Symmetry Channel (BSC), discrete memoryless channel.'
+
+features:            
+  decoders:
+    - 'Ball-Collision Decoding \cite{doi:10.1007/978-3-642-22792-9_42}'
+    - 'Finiasz and Sendrier (FS-ISD Decoding)\cite{doi:10.1007/978-3-642-10366-7_6}'
+             
+realizations:
+    - 'Distributed Storage Systems \cite{doi: 10.1109/TPDS.2008.251}'
+  
+notes:
+    - 'Given N and the rate R, the minimum distance of a typical random code (TRC) is given by the Gilbert–Varshamov (GV) bound d=N\delta_{GV}(2R).
```

Comment:
First note --> protection field, adding defn of h^{-1}. Second note can be merged into rate, although more specific random-code-related info is needed there.